glom/core.py

```diff
@@ -1323,13 +1323,35 @@ def _t_child(parent, operation, arg):
     return t
 
 
+def _s_first_magic(scope, key, _t):
+    """
+    enable S.a to do S['a'] or S['a'].val as a special
+    case for accessing user defined string variables
+    """
+    pae = None
+    try:
+        cur = scope[key]
+    except (KeyError, IndexError, TypeError) as e:
+        pae = PathAccessError(e, Path(_t), 0)
+    if isinstance(cur, _VarVal):
+        cur = cur.val
+        if cur is _UNDEFINED:
+            raise GlomError('var referenced before assignment')
```

Comment:
I like this, might deserve its own error type at some point, but at the very least the var name should be in here.

Reply:
100% agree.... this was the same-day special it needs polish